### DISCUSSION TITLE
Is there a way to use Lucide icons in Figma like Font Awesome?

### DISCUSSION THREAD

**Discussion by fernandordev** [lucide-icons/lucide#2553] [Help]
Hey everyone! I’m using the Font Awesome component in Figma, and it's great because you just type the icon name, and it appears right away – super handy for creating Figma components. I’m trying to do the same with the Lucide icons font, but I can’t seem to get it working. Does anyone know how to set it up?

For example, with Font Awesome, if I type "smile" using the fw-font, the smile icon appears instantly. I'd like to do the same thing with Lucide. The only way I've managed to get it working is by copying the Unicode directly from the page file, but that’s not exactly a convenient solution. A Unicode copy option on lucide.dev would be super helpful!

https://www.figma.com/community/file/994237671938398148/font-awesome-official-icon-component-free-pro

**Comment by fernandordev** [comment]
Got it working! It’s not the most polished implementation, but it does the job for now. I’m figuring out how to make it more like Font Awesome.

https://www.figma.com/community/file/1435408799047969963/lucide-icons-with-unicode-font

**Reply by WalleyeAU** [reply]
> Got it working! It’s not the most polished implementation, but it does the job for now. I’m figuring out how to make it more like Font Awesome.
> 
> https://www.figma.com/community/file/1435408799047969963/lucide-icons-with-unicode-font

Hey @fernandordev did you ever manage to move this forward? I used FontAwesome a lot so that I could tokenise (through Token Studio) my icons, centralising the source of truth and using these tokens to drive both Figma icon components and also a custom React component. I wrote up an article on it here: https://medium.com/tokens-studio/how-to-apply-design-tokens-to-your-icons-1439fd30061f

With your approach, the issue I have is that the unicode value isn't easy to understand, you can't read it and I don't think pass it through into a JSON to be read. In an ideal world, I'd be able to do the same approach I had done with FontAwesome, but with Lucide Icons instead.

**Reply by fernandordev** [reply]
Hello @WalleyeAU , at the time I found that it was too much work to make it like Font Awesome, so I dropped the idea. The solution I proposed worked fine for me. I’m not designing much lately, sorry haha.

**Reply by WalleyeAU** [reply]
All good @fernandordev! I was wondering if you might help with something then. Since your work on this. I think some of the cods have changed as the icons that get displayed do not correspond the the icon that is rendered. No big deal, but I'm unsure how you get the unicode for each icon to work with the Lucide font

For example, if I look at the unicode html file that comes with the Lucide font, and I create a text layer in Figma with this font applied, when i copy and paste in the unicode, nothing happens to the text.

Example: I want to use the icon `a-arrow-down`. The code for this is `&#58757;`, however this doesn't render the glyph in Figma.

Any pointers on how to use these unicodes would be very welcome!

**Reply by fernandordev** [reply]
Hello @WalleyeAU 

Back then, I created a script [lucide-figma-icons.js](https://github.com/user-attachments/files/24571314/lucide-figma-icons.js) to generate a JSON file that can be imported into Figma as variables. Maybe what’s happening on your side is that the font isn’t updated. The latest version of the font should match the Unicode correctly.

What I did back then was take this [unicode.html](https://github.com/user-attachments/files/24571423/unicode.html) file and copy-paste it into my script to generate the correct JSON. Then, I imported it into Figma using an extension for that.

The variables are just to make it easier to find the icon by name, because what should actually work is installing the font and pasting the Unicode there. 

<img width="1012" height="379" alt="image" src="https://github.com/user-attachments/assets/a7ca03dd-d3f1-4f81-935c-ba43ba4791f5" />


 <- this is the unicode, not the &#58757, you need to copy the icon not the code

<img width="1265" height="672" alt="image" src="https://github.com/user-attachments/assets/09052f42-69cd-4c41-ba9f-36de5cef794a" />
